account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py

```diff
@@ -14,32 +14,32 @@ class AccountFiscalPosition(models.Model):
     )
 
     @api.model
-    def search(self, args, offset=0, limit=None, order=None, count=False):
-        if "vat_vies_required" in self.env.context:
-            args = expression.AND(
+    def search(self, domain, offset=0, limit=None, order=None, count=False):
+        if self.env.context.get("vat_vies_required"):
+            domain = expression.AND(
                 (
-                    args,
+                    domain,
                     [
                         (
                             "vat_vies_required",
                             "=",
-                            self.env.context.get("vat_vies_required"),
+                            self.env.context["vat_vies_required"],
                         )
                     ],
                 )
             )
         return super().search(
-            args, offset=offset, limit=limit, order=order, count=count
+            domain, offset=offset, limit=limit, order=order, count=count
         )
 
     @api.model
-    def get_fiscal_position(self, partner_id, delivery_id=None):
+    def _get_fiscal_position(self, partner, delivery=None):
         _self = self
-        if partner_id:
-            partner = self.env["res.partner"].browse(delivery_id or partner_id)
+        if partner:
+            partner = partner or delivery
```

Comment:
It should have more priority delivery than partner.

Reply:
Changed!